src/pointer/inner.rs

```diff
@@ -422,7 +522,7 @@ impl<'a> PtrInner<'a, [u8]> {
         // `validate_cast_and_convert_metadata`. Thus, this call to
         // `validate_cast_and_convert_metadata` will only panic if `U` is a DST
         // whose trailing slice element is zero-sized.
-        let maybe_metadata = layout.validate_cast_and_convert_metadata(
+        let maybe_metadata = crate::util::MetadataOf::<U>::validate_cast_and_convert_metadata(
```

Comment:
This change is invalid - we sometimes *want* to use the synthesized `layout` (namely when the argument `meta` is `Some`).